### ISSUE TITLE
runCauldronContainerGen failed: TypeError: this.stream.clearLine is not a function

### ISSUE THREAD

**Issue by jomnius** [electrode-io/electrode-native#666]
What could be wrong (except bad error note), when command

```bash
ern create-container --descriptor MyDemo:android:0.0.1
```

...fails with this error:

> 16:19:10 runCauldronContainerGen failed: TypeError: this.stream.clearLine is not a function
16:19:10 An error occurred: this.stream.clearLine is not a function

When I run the same thing with debug level trace, I got some different info:

> 15:40:22 Starting a Gradle Daemon (subsequent builds will be faster)
15:40:28 Configuration 'androidTestCompile' in project ':app' is deprecated. Use 'androidTestImplementation' instead.
15:40:28 Configuration 'compile' in project ':app' is deprecated. Use 'implementation' instead.
15:40:28 Configuration 'testCompile' in project ':app' is deprecated. Use 'testImplementation' instead.
15:40:29 :app:preBuild UP-TO-DATE
15:40:31 :app:preDebugBuild FAILED
15:40:31 
15:40:31 FAILURE: Build failed with an exception.
15:40:31 
15:40:31 * What went wrong:
15:40:31 Could not resolve all files for configuration ':app:debugCompileClasspath'.
15:40:31 > Could not find com.walmartlabs.ern:runner-ern-container:1.0.0.
15:40:31   Searched in the following locations:
15:40:31       file:/usr/local/share/android-sdk/extras/m2repository/com/walmartlabs/ern/runner-ern-container/1.0.0/runner-ern-container-1.0.0.pom
15:40:31       file:/usr/local/share/android-sdk/extras/m2repository/com/walmartlabs/ern/runner-ern-container/1.0.0/runner-ern-container-1.0.0.jar
15:40:31       file:/usr/local/share/android-sdk/extras/google/m2repository/com/walmartlabs/ern/runner-ern-container/1.0.0/runner-ern-container-1.0.0.pom
15:40:31       file:/usr/local/share/android-sdk/extras/google/m2repository/com/walmartlabs/ern/runner-ern-container/1.0.0/runner-ern-container-1.0.0.jar
15:40:31       file:/usr/local/share/android-sdk/extras/android/m2repository/com/walmartlabs/ern/runner-ern-container/1.0.0/runner-ern-container-1.0.0.pom
15:40:31       file:/usr/local/share/android-sdk/extras/android/m2repository/com/walmartlabs/ern/runner-ern-container/1.0.0/runner-ern-container-1.0.0.jar
15:40:31       file:/Users/jenkins/.m2/repository/com/walmartlabs/ern/runner-ern-container/1.0.0/runner-ern-container-1.0.0.pom
15:40:31       file:/Users/jenkins/.m2/repository/com/walmartlabs/ern/runner-ern-container/1.0.0/runner-ern-container-1.0.0.jar
15:40:31       https://jcenter.bintray.com/com/walmartlabs/ern/runner-ern-container/1.0.0/runner-ern-container-1.0.0.pom
15:40:31       https://jcenter.bintray.com/com/walmartlabs/ern/runner-ern-container/1.0.0/runner-ern-container-1.0.0.jar
15:40:31       https://dl.google.com/dl/android/maven2/com/walmartlabs/ern/runner-ern-container/1.0.0/runner-ern-container-1.0.0.pom
15:40:31       https://dl.google.com/dl/android/maven2/com/walmartlabs/ern/runner-ern-container/1.0.0/runner-ern-container-1.0.0.jar
15:40:31   Required by:
15:40:31       project :app
15:40:31 
15:40:31 * Try:
15:40:31 Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.
15:40:31 
15:40:31 * Get more help at https://help.gradle.org

My best guess: it's related to access rights i.e. build script was not able to download something? Maybe? For example first mentioned path does not exist, nor /Users/jenkins/.m2 folder. But .m2 shouldn't necessarily exist yet?

> mac-mini-jenkins-5:extras jenkins$ cd /usr/local/share/android-sdk/extras/
mac-mini-jenkins-5:extras jenkins$ ls -l
total 0
drwxr-xr-x  3 jenkins  admin  96 Mar  8 01:29 google
mac-mini-jenkins-5:extras jenkins$ 

Btw my node should be new enough:

```bash
mac-mini-jenkins-5:workspace jenkins$ node -v
v9.3.0
```

**Comment by belemaire** [comment]
@jomnius 
We have encountered this issue on our CI env as well.
It is due to Ora (the JS library we use to display spinners during commands execution) that doesn't properly work in certain CI envs. 
Anyway, spinners in the terminal are not really useful on CI.
One way to get rid of this error is to activate `debug` or `trace` log level of Electrode Native. For such log level we disable the spinners. Independently of the CI env related issue, even when running in user terminal spinners are kind of messing with verbosity of such log levels, so we just disable them.
We are using `trace` level on CI. While it's quite verbose, we want as much info as possible in case a command is failing. It also has the benefit of disabling the spinners, thus solving the `stream.clearLine` error.

There are two ways to switch to `trace` log level in Electrode Native :

1) If you want to enable a given log level for a single command run, you can prefix the command with `ERN_LOG_LEVEL=trace`. For example `ERN_LOG_LEVEL=trace ern create-container ...`

2) If you want to enable a given log level for all commands, you can use `ern platform config logLevel trace`. This will set the log level in Electrode Native configuration file stored on the machine, so that any command will then use this log level (then you can run command normally `ern create-container ...`)

**Comment by jomnius** [comment]
@belemaire Thanx for explaining the clearLine issue and how to fix it. Appreciated, will take into use! I believe I can reset debug back to default afterwards by using `ERN_LOG_LEVEL=info`?

```bash
#!/bin/sh
export  ERN_LOG_LEVEL=trace
```

...however if you read a bit further, I did include output I got when using `ERN_LOG_LEVEL=trace`. Sorry, if I didn't make that more clear :)

**Problem remains** and I do believe it's about missing/wrong access right needed by `create-container`, when it internally wants to access either master manifest from GitHub or (more likely) my cauldron from company enterprise GitHub.

Any ideas appreciated? I believe I've already tried GitHub access token with repo and admin rights (now in use) as well as SSH (experimenting). The first one solved GitHub status check visibility, doesn't seem to be related to second problem... Almost like I would need two different access rights? Should or should I not use jenkins ssh-agent plugin (experimenting)? Or would jenkins Authorization plugin help (experimenting)?

I've lost count of what I've already tried. Learning new stuff every day :)

**Comment by jomnius** [comment]
No, but you're totally right: you did both explain and solve the problem I reported. The output from trace is something totally different.

This ticket can be closed. Thanx!

**Comment by belemaire** [comment]
That's ok, just that usually tickets contain just one issue so I have overlooked ;P
I took a look indeed at the first error only for`this.stream.clearLine`. 
That's one issue ;) The other one is related to the Android native application project not being able to locate the Container AAR (maven artifact).
For this one, it seems like it's not access right, but more than the Container was not published (using `publish-container` command if you are using `create-container` without relying on a Cauldron) ... or that the specific remote publication repository (MavenLocal, Nexus or Jcenter are the options of EN for Android) was not added to the repositories list of the runner application project top level build.gradle. 
Will close the ticket, feel free to open another issue if you still have issue around publication of the Container for android.
